### PR TITLE
Add event number filter to json:api event resource

### PR DESCRIPTION
**Pull request by simfeld**
cc @gexclaude

### CODE REVIEW

File: spec/resources/events/reads_spec.rb [hunk 1]
```diff
@@ -124,6 +124,23 @@
...
+      let!(:one) { Fabricate(:event, number: 'COURSE-123') }
+      let!(:two) { Fabricate(:event, number: 'COURSE-456') }
+
+      it 'returns only events with equal number' do
+        params[:filter] = { number: course.number }
+        render
+        expect(jsonapi_data).to have(1).items
+      end
+
+      it 'returns only events wih number having a specific prefix' do
```
Comment (simfeld) [inline spec/resources/events/reads_spec.rb:137]:
```suggestion
      it 'returns only events with number having a specific prefix' do
```

File: spec/support/resource_spec_helper.rb [hunk 2]
```diff
@@ -10,7 +10,7 @@ module ResourceSpecHelper
 
   included do
     let(:ability) { Ability.new(person) }
-    let(:person) { people(:top_leader) }
+    let(:person) { Fabricate(:person) }
```
Comment (simfeld) [inline spec/support/resource_spec_helper.rb:13]: See https://github.com/simfeld/hitobito_pbs/actions/runs/8669188064/job/23775501581
```
Failure/Error: raise StandardError, "No fixture named '#{f_name}' found for fixture set '#{fs_name}'"

     StandardError:
       No fixture named 'top_leader' found for fixture set 'people'
```

**Review by kronn — APPROVED** [review]

**Comment by kronn** [comment]
I tried to rebase this PR so it can be merged. Unfortunately, the number is usually not an attribute on events. That's why it doesn't work that way. Either this would have to be solved directly in the wagon or only for courses. But I could not do it quickly.

I generally think filtering by number is a good idea. Could you update this PR accordingly?

**Comment by kronn** [comment]
Nevermind, I found the problem I had, will resolve it and merge it.

Events normally do not have a number, so the attribute currently is only present on courses. When moving the change and the specs to the course-part, it works like it should.

**Comment by simfeld** [comment]
@kronn thanks for finding the proper fix 🚀

**Pull request closed without merging**